moto/cloudformation/models.py

```diff
@@ -346,10 +346,11 @@ def update(
                 instance.parameters = parameters or []
 
     def delete(self, accounts: List[str], regions: List[str]) -> None:
-        for i, instance in enumerate(self.stack_instances):
+        for instance in self.stack_instances:
             if instance.region_name in regions and instance.account_id in accounts:
                 instance.delete()
-                self.stack_instances.pop(i)
+        # clear the stack_instances in the end to prevent IndexError
+        self.stack_instances.clear()
```

Comment:
This isn't quite right. See the failing test - we should only remove the stack instances that match the provided regions/accounts

Reply:
Thanks for the feedback , that makes sense - will look into alernative fix.